apps/web/src/pages/[document].tsx

```diff
@@ -147,6 +156,55 @@ const DocumentPage: NextPage<Props> = ({ page: initialPage }) => {
     }
   );
 
+  const deletePageMutation = useMutation(
+    async () => {
+      const ceramicSession = await ceramic.getSession();
+
+      const response = await fetch("/api/page/unpublish", {
+        method: "POST",
+        body: JSON.stringify({
+          pageId: page!.id,
+          ceramicSession: ceramicSession?.serialize(),
+        }),
+      });
+
+      if (!response.ok) {
+        throw new Error(`${response.status} ${response.statusText}`);
+      }
+
+      await response.json();
+
+      trackEvent("Page Unpublished");
+
+      return await deletePage(page!.id);
```

Comment:
What do you think about adding a try-catch block here to be more specific where the error stems from? Then it can be propagated to the `useMutation` function.

Something like: 
```
try {
      return await deletePage(page!.id);
    } catch (error) {
      console.error('Error deleting the page:', error);
      throw error;
    }
    ```